From ca1b411c3d1c3ef2abd1c39268fb8ef8c25c4614 Mon Sep 17 00:00:00 2001
From: Jayachandran B <jayachandran.b@intel.com>
Date: Mon, 22 Sep 2014 23:15:48 +0530
Subject: [PATCH 2/7] ASoC: dapm: Fix debug print in soc-dapm.c

Issue: GMINL-1602

In snd_soc_dapm_connect_dai_link_widgets(), a debug print tried to
access the pointers cpu_dai->codec and codec_dai->platform. These
are null and hence would cause panic if this debug statement
is enabled. Fix it by correct pointers cpu_dai->platform and
codec_dai->codec

Change-Id: I59365b2b866f69e9b7b6cfee28f90e3104e06149
Signed-off-by: Jayachandran B <jayachandran.b@intel.com>
---
 sound/soc/soc-dapm.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/soc-dapm.c b/sound/soc/soc-dapm.c
index 3b63612..417686e 100644
--- a/sound/soc/soc-dapm.c
+++ b/sound/soc/soc-dapm.c
@@ -3745,8 +3745,8 @@ void snd_soc_dapm_connect_dai_link_widgets(struct snd_soc_card *card)
 			r.source = cpu_dai->playback_widget->name;
 			r.sink = codec_dai->playback_widget->name;
 			dev_dbg(rtd->dev, "connected DAI link %s:%s -> %s:%s\n",
-				cpu_dai->codec->name, r.source,
-				codec_dai->platform->name, r.sink);
+				cpu_dai->platform->name, r.source,
+				codec_dai->codec->name, r.sink);
 
 			snd_soc_dapm_add_route(&card->dapm, &r, true);
 		}
-- 
1.7.9.5

